nesta/packages/crunchbase/crunchbase_collect.py

```diff
@@ -0,0 +1,333 @@
+from contextlib import contextmanager
+import logging
+import pandas as pd
+import re
+import requests
+import tarfile
+from tempfile import NamedTemporaryFile
+
+from nesta.production.luigihacks import misctools
+from nesta.production.orms.orm_utils import insert_data
+from nesta.packages.geo_utils.country_iso_code import country_iso_code_to_name
+from nesta.packages.geo_utils.geocode import generate_composite_key
+
+
+@contextmanager
+def crunchbase_tar():
+    """Downloads the tar archive of Crunchbase data.
+
+    Returns:
+        :code:`tarfile.Tarfile`: opened tar archive
+    """
+    crunchbase_config = misctools.get_config('crunchbase.config', 'crunchbase')
+    user_key = crunchbase_config['user_key']
+    url = 'https://api.crunchbase.com/v3.1/csv_export/csv_export.tar.gz?user_key='
+    with NamedTemporaryFile() as tmp_file:
+        r = requests.get(''.join([url, user_key]))
+        tmp_file.write(r.content)
+        try:
+            tmp_tar = tarfile.open(tmp_file.name)
+            yield tmp_tar
+        finally:
+            tmp_tar.close()
+
+
+def get_csv_list():
+    """Gets a list of csv files within the Crunchbase tar archive.
+
+    Returns:
+        list: all .csv files int the archive
+    """
+    csvs = []
+    csv_pattern = re.compile(r'^(.*)\.csv$')
+    with crunchbase_tar() as tar:
+        names = tar.getnames()
+    for name in names:
+        tablename = csv_pattern.match(name)
+        if tablename is not None:
+            csvs.append(tablename.group(1))
+    return csvs
+
+
+def get_files_from_tar(files, test=False):
```

Comment:
Should `nrows` be an argument rather than `test`?

Reply:
@jaklinger with test as the kwarg I can just supply the luigi task's `self.test` variable. If I change it to nrows I'll need to perform `nrows = 1000 if self.test else None` each time before calling the function.

Reply:
test mode removed from package